Backend/src/main/java/ktsnvt/tim1/repositories/ReservationRepository.java

```diff
@@ -1,12 +1,25 @@
 package ktsnvt.tim1.repositories;
 
 import ktsnvt.tim1.model.Reservation;
+
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.Date;
+import java.util.List;
 
 public interface ReservationRepository extends JpaRepository<Reservation, Long> {
     Page<Reservation> findByOrderIdIsNull(Pageable pageable);
 
     Page<Reservation> findByOrderIdIsNotNull(Pageable pageable);
+
+    @Query(value = "select ed.date, count(t), sum(esg.price) from Reservation r join r.tickets t join t.eventDays ed " +
+            "join t.eventSeatGroup esg join ed.event e join e.location l where r.orderId is not null and ed.date " +
```

Comment:
Event object can be taken directly from reservation r.
Change `join ed.event e` to `join r.event e`